src/components/Search/CompareWithBase.tsx

```diff
@@ -137,25 +132,22 @@ function CompareWithBase({
     expandBaseComponent(true);
   };
 
-  const handleCancelBase = () => {
-    setInProgressBaseRev(baseStagingRev);
-  };
-  const handleCancelNew = () => {
-    setInProgressNewRevs(newStagingRevs);
-  };
-
-  const handleSaveBase = () => {
-    setStagingBaseRev(baseInProgressRev);
-  };
-  const handleSaveNew = () => {
-    setStagingNewRevs(newInProgressRevs);
+  const handleCancel = () => {
+    setInProgressNewRevs(newRevs);
+    setInProgressBaseRev(baseRev);
+    setFormIsDisplayed(false);
+    setHasCancelButton(false);
```

Comment:
I believe we also need to set it to false when the form is submitted. Curiously this didn't look so strange with the previous design, but now I really feel like it's needed.

My suggestion therefore is to:
1. rename `possiblyPreventFormSubmission` to `onFormSubmit`
2. In that function, add the code to hide the form.

Tell me what you think!

Reply:
Yes, this makes sense. I thought of this as well.